DOCS-2765: Add installation guide for MKE 4k (#2441)
tigera/docs: calico-enterprise/installation/install-calico-enterprise-mke-4k.mdx

## Start the MKE 4k installation

1. In your installation configuration file, make the following changes:
   1. Specify that you want to use a custom networking provider:
      ```yaml
      spec:
        network:
          providers:
            - enabled: true
            provider: "custom"
      ```
   1. Optional: If you don't want to use the default pod CIDR range 192.168.0.0/16, specify the range in the installation configuration file.
      The pod CIDR range cannot easily be changed later, and it must be configured before you install $[prodname].
      Edit the configuration file as follows:
      ```yaml
      spec:
        network:
          providers:
            - enabled: true
            provider: "custom"
            extraConfig:
              cidrV4: <pod-cidr>
      ```
      Replace `<pod-cidr>` with a pod CIDR range that suits your deployment and does not conflict with other ranges in your network.

1. To start the MKE 4k installation process, run the following command:

   ```bash
   mkectl apply -f <mke-configuration-file>.yaml --cni-check-timeout <cni-timeout-duration>
   ```
   Replace the following:

>    * `<mke-configuration-file>`: The name of your installation configuration file.

Reviewer: ```suggestion
   * `<mke-configuration-file>`: The path of your MKE4 installation configuration file.
```

Author: Thanks, fixed.

>    * `<cni-timeout-duration>`: The amount of time, in minutes, that the MKE 4k installer will wait for you to complete the network installation.

Reviewer: ```suggestion
   * `<cni-timeout-duration>`: The duration in minutes, that the MKE 4k installer will wait for you to complete the CNI installation.
```

> 

Reviewer: Should we add a side note that -
You can create the MKE4 installation config file by using -
```bash
mkectl init > mke4.yaml
```

Author: I felt that this detail was unnecessary, and clearly part of the MKE side of things. What's the use in telling people that small detail, when we don't offer guidance on the far more difficult tasks? Like setting up infra, getting that config to sync with machines you set up, etc.

Could go either way.

   Wait until you see that the installer is waiting for cluster networking to be established.

   ```bash title="Example output"
   waiting 60 minutes for cluster networking to be established using custom CNI provider
   ```

1. SSH into one of the controller nodes and run the following command:

   ```bash
   sudo /usr/local/bin/k0s kubeconfig admin
   ```

   Add the kubeconfig to your local kubectl.

   :::important
   Don't close this terminal session as you move to the next step.

>    The MKE 4k installation process does until after you install $[prodname].

Reviewer: ```suggestion
   The MKE 4k installation continues on this thread after you install $[prodname].
```

Author: Ugh, thanks.

   For the next step, open a second terminal session.
   After you install $[prodname], you can return to the first terminal session to observe and verify the installation process.
   :::

## Install Calico Enterprise

In a new terminal, install the $[prodname] CNI.

1. [Configure a `StorageClass` resource for ElasticSearch](../operations/logstorage/create-storage.mdx).

1. Install the Tigera Operator and custom resource definitions.

   ```bash

>    kubectl create -f $[filesUrl]/manifests/operator-crds.yaml

Reviewer: This has to be applied server-side else the size of the crd makes k8s unhappy (client validation throws the error about the size of the file being greater)-
```bash
kubectl apply --server-side -f $[filesUrl]/manifests/operator-crds.yaml
```

Author: Done, thanks.

We currently include this only for upgrades in CE. Should this command be used more generally?
